### PR TITLE
Add data processing tools

### PR DESCRIPTION
**Pull request by kevindougherty-noaa**
This PR adds data processing tools needed to create intermediate .nc files with information for plotting things like counts, averages, standard deviations etc.

### CODE REVIEW

**Comment by kevindougherty-noaa** [comment]
There are some areas I believe can be improved, but will take advantage of the systems down tomorrow to have you all take a look and see what you think.

File: parm/gfs/coefb_atms_n20.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# Data read
+# ---------
+datasets:
+  - name: bcoef
+    satellite: atms
+    sensor: n20
+    type: RadDataSpace
+    filename: /path/to/diag_atms_n20_2021032500.nc
+    channels: [1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22]
+    groups:
```
Comment (CoryMartin-NOAA) [inline parm/gfs/coefb_atms_n20.yaml:12]: As we chatted about, perhaps something more like this:
```
variables to process:
- function: calculate_mean_omf
  variables: [brightnessTemperature]
  channels: &chans
  groups: [hofx0, hofx1]
  bias corrected: false
  groups out: [omb0Mean, omb1Mean]
- function: calculate_mean
  variables: [brightnessTemperature]
  channels: &chans
  groups: [obsBias0, constant]
  groups out: [obsBiasMean, constantMean]
```

File: ush/biasCorrectionReader.py [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+
+    # Close the netCDF file
+    nc_file.close()
+
+    return
+
+
+def main(input_file, cycle, satellite, channels, outvars, outfile, group_names, group_variable):
+    """
+    Read in config file contaiting bias corrected data, calculate counts, averages, and standard
```
Comment (EdwardSafford-NOAA) [inline ush/biasCorrectionReader.py:146]: typo in 'containing'.

**Comment by kevindougherty-noaa** [comment]
Closing this PR as changes are going to be pretty significant.

**Pull request closed without merging**